### PR TITLE
Add support for reading secrets across region

### PR DESCRIPTION
**Pull request by zsims**
This adds support for reading secrets from a region other than the current one.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,28 @@ steps:
               secret-id: 'arn:aws:secretsmanager:ap-southeast-2:1234567:secret:my-global-file-secret'
 ```
 
+## For Secrets in Another Region
+
+In order to request a secret from another region from your agents, a region parameter must be provided.
+This is independent of the secret ARN, which may confusingly include a region.
```
Comment (mr-beerkiss) [inline README.md]: > which may confusingly include a region

The way this is worded makes it seem you could specify a different region to that contained in the ARN.  Would that even be possible to have as parameters and still return a valid value?

**Review by mr-beerkiss — APPROVED** [review]
Looks good to me.  Just a question about some wording in the documentation which is more out of curiosity than out of genuine concern.

File: lib/shared.bash [hunk 2]
```diff
@@ -7,6 +7,12 @@ function strip_quotes() {
 function get_secret_value() {
   local secretId="$1"
   local allowBinary="${2-}"
+  local region="${3-}"
```
Comment (ashlineldridge) [inline lib/shared.bash]: Should these have `:`'s? I.e. `region=${3:-}`?

File: README.md [hunk 3]
```diff
@@ -68,6 +68,28 @@ steps:
...
+
+```yml
+steps:
+  - commands: 'echo \$MY_LOCAL_SECRET and \$MY_GLOBAL_SECRET'
+    plugins:
+      - seek-oss/aws-sm#v0.0.5:
+          env:
+            - secret-id: 'arn:aws:secretsmanager:us-west-2:1234567:secret:my-global-secret'
+              env-name: MY_GLOBAL_SECRET
+              region: us-west-2
```
Comment (ashlineldridge) [inline README.md]: Couldn't you infer the region from the ARN the same way we do in `secretsmanager.py`?

Comment (zsims) [reply README.md]: That's a much better idea, and also addresses Darren's point. Will update.

File: README.md (hunk 1 above)
Comment (zsims) [reply README.md]: That's a very good point, and makes this rather error prone. I'll update the PR to infer the region from the ARN.

File: lib/shared.bash (hunk 2 above)
Comment (zsims) [reply lib/shared.bash]: They can, the difference is null vs empty handling. I'll update 😁

**Comment by zsims** [comment]
Updated to deduce the region from the ARN, thanks for the suggestion @mr-beerkiss @ashlineldridge  :-)

**Pull request merged**